File: transformation/src/main/java/org/opensearch/migrations/Flavor.java
package org.opensearch.migrations;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public enum Flavor {
    ELASTICSEARCH("ES"),
    OPENSEARCH("OS"),
    AMAZON_MANAGED_OPENSEARCH("AOS"),
    AMAZON_SERVERLESS_OPENSEARCH("AOSS");

    final String shorthand;

    public boolean isOpenSearch() {
        switch (this) {
            case OPENSEARCH:
            case AMAZON_MANAGED_OPENSEARCH:
            case AMAZON_SERVERLESS_OPENSEARCH:
                return true;
            default:
                return false;
        }
    }
}

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/flavor.py
from enum import Enum


class Flavor(str, Enum):
    ELASTICSEARCH = "elasticsearch"
    OPENSEARCH = "opensearch"
    
    @property
    def shorthand(self) -> str:
        """Return a shorthand representation of the flavor"""
        if self == Flavor.ELASTICSEARCH:
            return "es"
        elif self == Flavor.OPENSEARCH:
            return "os"
        return ""
